sources/api/controllers/UserController.ts

```diff
@@ -96,20 +100,73 @@ export async function createUser(req: Request, res: Response) {
     if (!req.body.pseudo || !req.body.password || !req.body.email)
         throw new ApiException(400, "Invalid request");
 
-    const user: User = new User(
+    // Check if user exists before creating PendingUser
+    const connection = await DbConnection.getConnection();
+    const UserRepository = connection.getRepository(User);
+    const peudoExists = await UserRepository.findOne({where : {pseudo: req.body.pseudo }})
+    const emailExists = await UserRepository.findOne({where : {email: req.body.email }})
+    
+    if (peudoExists || emailExists) {
+        throw new ApiException(400, "User already exists");
+    }
+
+    const validation_link: string = AccountVerification.generateLink();
+    const user: PendingUser = new PendingUser(
         req.body.pseudo,
         req.body.email,
-        await User.cipherPassword(req.body.password)
+        await PendingUser.cipherPassword(req.body.password),
+        validation_link
     );
     if (!user.isValid())
         throw new ApiException(400, "Invalid fields in User");
-
+    
     // Save the created User
     const conn = await DbConnection.getConnection();
     await conn.manager.save(user);
 
+    var reqUrl = getReqUrl(req)
+    if (!reqUrl.endsWith('/')) {
+    	reqUrl += '/';
+    }
+
+    const link = reqUrl + 'validate/' + validation_link;
+    EmailSender.sendEmail(user.email, 'Account Verification', link);
+
+    if (process.env.SENDEMAIL) {
+        return res.status(201).json({
+            message: "A link has been sent to your email address, please click on it in order to confirm you email",
+            user: UserView.formatPendingUser(user)
+        });
+    }
+    return res.status(201).json({
+        message: "A link has been sent to your email address, please click on it in order to confirm you email",
+        link: link,
+        user: UserView.formatPendingUser(user)
+    });
```

Comment:
Code dupplication here

Create an intermediary `response_body` and add the field `link` inside if the condition is respected